lib/web3-contracts.js

```diff
@@ -131,248 +106,189 @@ export function useTokenBalance(symbol) {
+      // 2. Open a buy order (and wait for the transaction to be mined)
+      // 3. Claim the order (done in useClaimOrder)
+
+      // We first check for allowance if the direction is correct,
+      // and if we need more, approve
+      if (forwards) {
+        const allowance = await antContract.allowance(
+          account,
+          marketMakerAddress
+        )
+        if (allowance.lt(bigNum(amount))) {
+          try {
+            await antContract.approve(marketMakerAddress, amount)
+            // Don't wait for the approval to be mined before showing second transaction
+          } catch (err) {
+            throw new Error('User did not approve transaction')
+          }
+        }
+      }
 
-      return antContract.approveAndCall(wrapperAddress, antAmount, '0x00', {
-        gasLimit: 1000000,
-      })
+      // Then, we open a buy or sell order. The next step will be handled after the
+      // transaction gets mined, in useClaimOrder()
+      return await (forwards
+        ? fundraisingContract.openBuyOrder(antAddress, amount, {
+            gasLimit: 850000,
+            value: 0,
+          })
+        : fundraisingContract.openSellOrder(antAddress, amount, {
+            gasLimit: 850000,
+          }))
     },
-    [antContract, wrapperAddress]
+    [account, antAddress, antContract, fundraisingContract, marketMakerAddress]
   )
 }
 
-export function useConvertTokenToAnj(selectedToken) {
-  const { account } = useWeb3Connect()
-  const tokenContract = useKnownContract(`TOKEN_${selectedToken}`)
-  const wrapperContract = useKnownContract(`WRAPPER`)
-  const [tokenAddress] = getKnownContract(`TOKEN_${selectedToken}`)
-  const [wrapperAddress] = getKnownContract('WRAPPER')
+export function useClaimOrder() {
+  const { account, ethersProvider } = useWeb3Connect()
+  const fundraisingContract = useKnownContract('FUNDRAISING')
+  const [antAddress] = getKnownContract('TOKEN_ANT')
 
   return useCallback(
-    async (amount, estimatedAnj, activate = true) => {
-      if ((!tokenContract && selectedToken !== 'ETH') || !wrapperAddress) {
-        throw new Error('Could not get the token and wrapper contract.')
-      }
-
-      // now + 60s * 120min
-      const twoHourExpiry = Math.floor(Date.now() / 1000) + 60 * 120
-      const minAnj = calculateSlippageAmount(estimatedAnj).toString()
-
-      // As minAnj already ensures the amount of tokens received,
-      // we can safely ignore this value by setting it to 1
-      const minEth = bigNum(1)
-      // If the user has selected ETH, we can just send the ETH to the function
-      if (selectedToken === 'ETH') {
-        return await wrapperContract.contributeEth(
-          minAnj,
-          twoHourExpiry,
-          true,
-          {
+    async (transactionHash, forwards = true) => {
+      const { blockNumber } = await ethersProvider.getTransaction(
+        transactionHash
+      )
+      // We claim the buy order, with the blockNumber of the emitted open order
```

Comment:
I'd rename `blockNumber` to `batchId` as the proper concept; we get this "shortcut" with the block number because we have one block batch sizes.